lib/feeder_importer.rb

```diff
@@ -0,0 +1,95 @@
+require 'prx_access'
+require 'addressable/uri'
+
+class FeederImporter
+  include PRXAccess
+  include Announce::Publisher
+
+  attr_accessor :account_id, :user_id, :feeder_podcast_url
+  attr_accessor :podcast, :series, :template, :distribution
+
+  def initialize(account_id, user_id, feeder_podcast_url)
+    self.account_id = account_id
+    self.user_id = user_id
+    self.feeder_podcast_url = feeder_podcast_url
+  end
+
+  def import
+    retrieve_podcast
+    create_series
+  end
+
+  def create_series(pcast = podcast)
+    attrs = {
+      app_version: PRX::APP_VERSION,
+      account_id: account_id,
+      creator_id: user_id,
+      title: pcast.title,
+      short_description: pcast.attributes[:subtitle],
+      description_html: pcast.attributes[:description]
+    }
+    self.series = Series.create!(attrs)
+
+    # Add images to the series
+    if podcast.attributes['itunes_image'] && podcast.itunes_image['url']
+      image = series.images.create!(
+        upload: clean_string(podcast.itunes_image['url']),
+        purpose: Image::PROFILE
+      )
+      announce_image(image)
+    end
+
+    if podcast.attributes['feed_image'] && podcast.feed_image['url']
+      image = series.images.create!(
+        upload: clean_string(podcast.feed_image['url']),
+        purpose: Image::THUMBNAIL
+      )
+      announce_image(image)
+    end
+
+    # Add the template and a single file template
+    self.template = series.audio_version_templates.create!(
+      label: 'Podcast Audio',
+      promos: false,
+      length_minimum: 0,
+      length_maximum: 0
+    )
+
+    num_segments = 1
+    episode = podcast.episodes.first
+    if episode.attributes['media']
+      num_segments = [episode.media.count, num_segments].max
+    end
+    num_segments.times do |x|
+      num = x + 1
+      template.audio_file_templates.create!(
+        position: num,
+        label: "Segment #{num}",
+        length_minimum: 0,
+        length_maximum: 0
+      )
+    end
+
+    self.distribution = Distributions::PodcastDistribution.create!(
+      url: feeder_podcast_url,
+      distributable: series,
+      audio_version_template: template
+    )
+
+    series
+  end
+
+  def retrieve_podcast(a_id = account_id, p_url = feeder_podcast_url)
```

Comment:
any particular reason for making the account id & podcast URL overrideable?

Reply:
& ditto for optional `podcast` parameter for `create_series` def

Reply:
yes, clearer to test if you can call the method without setting the state on the importer

Reply:
👍